File: pascal/hellofolks/c/hellofolks.c
/* Say hello to a group of people */

#include <stdio.h>

const char *names[] = {
      "Joseph"
    , "José"
    , "Josephine"
    , "Giuseppe"
    , "Iosephus"
    , "Josefina"
    , "Josephine"
};

int main(void) {
    int i;
    for (i = 0; names[i] != NULL; ++i) {
        printf("Hello, %s!\n", names[i]);
    }
    return(0);
}

File: pascal/hellofolks/hellofolks.hs
{- say hello to a list of people
-}

folks :: [String]
folks = ["Joseph"
        , "José"
        , "Josephine"
        , "Giuseppe"
        , "Iosephus"
        , "Josefina"
        , "Josephine"
        ]

sayhello :: String -> String
sayhello s = "Hello, " ++ s ++ "!"

greeting :: [String] -> String
greeting names = unlines $ map sayhello names

main :: IO()
main = do
    putStr $ greeting folks

